### PR TITLE
[Lake] Add lake tablet rowset

### PR DESCRIPTION
**Pull request by wyb**
## What type of PR is this：
- [ ] bug
- [x] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Problem Summary(Required) ：

Use lake rowset metadata and old segment.

### CODE REVIEW

File: be/src/storage/lake/rowset.cpp [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+    Status do_get_next(vectorized::Chunk* chunk, vector<uint32_t>* rowid) override {
+        return _iter->get_next(chunk, rowid);
+    }
+
+private:
+    vectorized::ChunkIteratorPtr _iter;
+};
+
+Rowset::Rowset(std::string group, TabletSchemaPtr tablet_schema, RowsetMetadataPtr rowset_metadata)
+        : _group(std::move(group)), _tablet_schema(tablet_schema), _rowset_metadata(rowset_metadata) {}
```
Comment (imay) [inline be/src/storage/lake/rowset.cpp:48]:
```suggestion
        : _group(std::move(group)), _tablet_schema(std::move(tablet_schema)), _rowset_metadata(std::move(rowset_metadata)) {}
```

File: be/src/storage/lake/rowset.cpp [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+public:
+    SegmentIteratorWrapper(vectorized::ChunkIteratorPtr iter)
+            : ChunkIterator(iter->schema(), iter->chunk_size()), _iter(std::move(iter)) {}
+
+    void close() override {
+        _iter->close();
+        _iter.reset();
+    }
+
+    virtual Status init_encoded_schema(vectorized::ColumnIdToGlobalDictMap& dict_maps) override {
```
Comment (sduzh) [inline be/src/storage/lake/rowset.cpp:27]:
```suggestion
   Status init_encoded_schema(vectorized::ColumnIdToGlobalDictMap& dict_maps) override {
```

File: be/src/storage/lake/rowset.cpp [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+        RETURN_IF_ERROR(vectorized::ChunkIterator::init_encoded_schema(dict_maps));
+        return _iter->init_encoded_schema(dict_maps);
+    }
+
+    virtual Status init_output_schema(const std::unordered_set<uint32_t>& unused_output_column_ids) override {
```
Comment (sduzh) [inline be/src/storage/lake/rowset.cpp:32]:
```suggestion
   Status init_output_schema(const std::unordered_set<uint32_t>& unused_output_column_ids) override {
```

**Review by sduzh — DISMISSED** [review]

**Review by sduzh — APPROVED** [review]

**Comment by wyb** [comment]
run starrocks_be_unittest

**Pull request merged**